### PR TITLE
Make version "algorithm" more Ruby-esque...

### PR DESCRIPTION
**Pull request by pvdb**
... fix a spelling error, and make the whole thing more scriptable for all other SB repos where we'll have to do a similar refactoring.

### CODE REVIEW

**Comment by lukaso** [comment]
@pvdb I'm worried that after this change tools like gem bump won't be able to correctly bump the version (which we will use in our automation). My vote would be to put this boiler plate at the end of the gemspec file and mutate the spec.version field.

**Comment by pvdb** [comment]
> I'm worried that after this change tools like gem bump won't be able to correctly bump the version (which we will use in our automation).

This PR just refactors what we already did in #14, so I assume your concerns are directed to those changes... but I get what you're saying.

When you say _"tools like gem bump"_ what exactly are referring to?

Are you referring to the `auto_bump_version.rb` script in some of our repositories?  If so, that script really needs some serious refactoring, though, and we can defo make it compatible.

Or do you have something else in mind?  Let me look at that first, and then cycle back to these changes... I feel like it'd be better to leave the gemspec alone, TBH, to avoid conflicts with future "gemspec sanitizer" changes.

**Comment by pvdb** [comment]
> My vote would be to put this boiler plate at the end of the gemspec file and mutate the spec.version field.

Not sure TBH: by doing that, things get inconsistent, which may not cause any issues today, but could very well start messing things up somewhere down the line...

```
$ GEM_PRE_RELEASE=blegga ruby -Ilib -r 'flip_fab/version' -r bundler -e '
puts format("gem itself: %s\n", FlipFab::VERSION); 
puts format("gemspec: %s\n", Bundler.load_gemspec("flip_fab.gemspec").version)'
gem itself: 1.0.3
gemspec: 1.0.3.blegga
$ _
```

Which one is it?

Lemme see if I can cook up a version that deals with all and everyone's concerns.

**Comment by pvdb** [comment]
@lukaso - please have a look at the refactoring I've done in c025d98

* it doesn't introduce a new constant
* it doesn't put functional logic in the gemspec
* it is Rubocop friendly
* it is "gem bump" friendly
* it is functionality equivalent to the original version _(it terms of BNW/Jenkins/gemstash)_
* but ... it also eliminates the "version discrepancy" I reported in an earlier comment
* and it is easy to script, to apply it to the other 40 repositories that need refactoring

So much win!

File: lib/flip_fab/version.rb [hunk 1]
```diff
@@ -1,10 +1,7 @@
...
+  VERSION = '1.0.5' # rubocop:disable Style/MutableConstant
 
-  # SB-specific versioning "algorithm" to accomodate Jenkins / gemstash
-  VERSION = if ENV['GEM_PRE_RELEASE'].nil? || ENV['GEM_PRE_RELEASE'].empty?
-                  FlipFab::GEM_VERSION
-                else
-                  "#{FlipFab::GEM_VERSION}.#{ENV['GEM_PRE_RELEASE']}"
-                end
+  # SB-specific versioning "algorithm" to accommodate BNW/Jenkins/gemstash
+  VERSION << '.' << ENV['GEM_PRE_RELEASE'].strip \
```
Comment (lukaso) [inline lib/flip_fab/version.rb]: Does changing a constant not create a warning? At least from rubocop?

Comment (lukaso) [reply lib/flip_fab/version.rb]: didn't see the rubocop disable.

Comment (pvdb) [reply lib/flip_fab/version.rb]: It's not really changing the `VERSION` constant, but rather the _(mutable)_ string object the constant is pointing at... nothing wrong with it from a Ruby point-of-view, but you're correct: Rubocop _(wrongly, IMO!)_ takes issue with it, which is why I added the `# rubocop:disable Style/MutableConstant` directive, to suppress the Rubocop warning.

Comment (pvdb) [reply lib/flip_fab/version.rb]:
```ruby
[1] pry(main)> BLEGGA = 'foo'
=> "foo"
[2] pry(main)> BLEGGA << ' ' << 'bar'
=> "foo bar"
[3] pry(main)> BLEGGA
=> "foo bar"
[4] pry(main)> BLEGGA = 'qux'
(pry):4: warning: already initialized constant BLEGGA
(pry):1: warning: previous definition of BLEGGA was here
=> "qux"
[5] pry(main)> _
```

**Pull request merged**